### ISSUE TITLE
Return the path in error handling

### ISSUE THREAD

**Issue by g-u-r-k-a-n** [dotnet/aspnetcore#35596]
I handle the errors very well throughout the project as described in [this section](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/error-handling?view=aspnetcore-5.0#usestatuscodepageswithredirects) of the document. Personally, I prefer just one controller for errors.

### Startup.cs 
```cs
app.UseStatusCodePagesWithRedirects("/Error?status={0}");
```
### ErrorController 
```cs
[ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)]
public IActionResult Index(int? status) {

    var errorViewModel = new ErrorViewModel {
        RequestId = Activity.Current?.Id ?? HttpContext.TraceIdentifier
    };

    var exceptionHandlerPathFeature = HttpContext.Features.Get<IExceptionHandlerPathFeature>();
   //handle the error
}
```
It returns the HTTP status code with `status` variable , but I want to return the previous request path as well. I was writing a new middleware from the `UseStatusCodePagesWithRedirects` extension, but the controller cannot catch the error because I changed the original pattern to another one. eg.`code={0}&path={1}` . I try different features in `HttpContext` but they only return status code normally. To put it simply, I want to handle referrer uri to the error controller.

# Feature Request 
It should be one more query string named for example `path` or something else and it shuold returns the previous uri in the `UseStatusCodePagesWithRedirects` extension.

Should be
```cs
app.UseStatusCodePagesWithRedirects("/MyStatusCode?code={0}&path={1}");
```
In this way, the path can handle in the controller as follows.
```cs
public IActionResult Index(int? status, string path) {
```

**Comment by Tratcher** [comment]
Note that UseStatusCodePages* handles more than error conditions, it would handle any response that did not provide a body. Responses without bodies are more common in Web API scenarios.

Have you tried using UseStatusCodePagesWithReExecute instead of UseStatusCodePagesWithRedirects? That behaves like UseExceptionhandler and preserves the request path in a feature as described here:
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/error-handling?view=aspnetcore-5.0#usestatuscodepageswithreexecute

**Comment by g-u-r-k-a-n** [comment]
@Tratcher Thanks for your suggestion. You are absolutely right, it was my fault  :see_no_evil: I had tried the 'UseStatusCodePagesWithRedirects' you pointed out before the post above, turns out I tried to handle it with type  `IExceptionHandlerPathFeature` and the future was normally `null`. I looked the paramaters again after your suggestion and I replaced it with `IStatusCodeReExecuteFeature` which is the correct type. Then future is not null and the `OriginalPath` is returns the path. 

![Debug View](https://i.stack.imgur.com/Z5CpW.png)

Thanks for your help :beers:

**Comment by g-u-r-k-a-n** [comment]
@Tratcher Actually, I've closed the issue, but may I ask a question? `UseStatusCodePagesWithReExecute` type has no an error property. If I use it, I'm unable to catch the exceptions this time. So, I want to catch the exceptions and to handle the OriginalPath as error type. I have created two action for this case.

```cs
app.UseStatusCodePagesWithRedirects("/Error/Code?code={0:int}");
app.UseStatusCodePagesWithReExecute("/Error/Status", "?status={0:alpha}");
```
But the pipeline doesn't allow for both as expected, the second extension overrides the first one and invokes only `Status` action, in otherwise the `Code` action. I hope I've made the point clear.

 So, in this case, I still need the path parameter I mentioned in the first message.
